File: src/sub_node_cpp/src/subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float32.hpp"

class SubscriberObject : public rclcpp::Node
{
public:
    SubscriberObject() : Node("cpp_subscriber")
    {
        steer_sub_ = this->create_subscription<std_msgs::msg::Float32>(
            "steer", 10, std::bind(&SubscriberObject::steer_callback, this, std::placeholders::_1));
        throttle_sub_ = this->create_subscription<std_msgs::msg::Float32>(
            "throttle", 10, std::bind(&SubscriberObject::throttle_callback, this, std::placeholders::_1));
    }

private:
    void steer_callback(const std_msgs::msg::Float32::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Received steer: %.2f", msg->data);
    }

    void throttle_callback(const std_msgs::msg::Float32::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Received throttle: %.2f", msg->data);
    }

    rclcpp::Subscription<std_msgs::msg::Float32>::SharedPtr steer_sub_;
    rclcpp::Subscription<std_msgs::msg::Float32>::SharedPtr throttle_sub_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SubscriberObject>());
    rclcpp::shutdown();
    return 0;
}

File: src/motor_controller/motor_controller/sub_node/subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import Float32
from piracer.vehicles import PiRacerPro

steer_alignment = -0.2
throttle_threshold = 0.4

class SubscriberObject(Node):
    def __init__(self):
        super().__init__('subscriber')

        self.piracer = PiRacerPro();

        self.create_subscription(Float32, 'steer', self.steer_callback, 10)
        self.create_subscription(Float32, 'throttle', self.throttle_callback, 10)

    def steer_callback(self, steer_msg):
        steer_value = steer_msg.data + steer_alignment
        self.get_logger().info(f'steer: {steer_value}')
        self.piracer.set_steering_percent(steer_value)

    def throttle_callback(self, throttle_msg):
        throttle_value = throttle_msg.data * throttle_threshold
        self.get_logger().info(f'throttle: {throttle_value}')
        self.piracer.set_throttle_percent(throttle_value)

def main(args=None):
    rclpy.init(args=args)
    node = SubscriberObject()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()